[coverage] Ignore pass using a regex

kinow opened the pull request:
Reverts #2578 

Sorry! :bow: 

**Check List**

- [x] I have read `CONTRIBUTING.md`.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [ ] Applied any dependency changes to `pyproject.toml`.
- [x] Tests are included (or explain why tests are not needed).
- [ ] Changelog entry included in `CHANGELOG.md` if this is a change that can affect users.
- [ ] Documentation updated.
- [ ] If this is a bug fix, PR should include a link to the issue (e.g. `Closes #1234`).

dbeltrankyl reviewed (approved):
Thanks @kinow don't worry!

LuiggiTenorioK commented:
In the API, I removed all the `pass` statements instead of ignoring them. In case it was for an empty function, I replaced the `pass` with a doc string.

```diff
  def empty_function():
-     pass
+    """ Explanation of what this function will be used for """
```

Kinda worked pretty good to increase coverage :sweat_smile:

kinow commented:
> In the API, I removed all the `pass` statements instead of ignoring them. In case it was for an empty function, I replaced the `pass` with a doc string.
> 
> ```diff
>   def empty_function():
> -     pass
> +    """ Explanation of what this function will be used for """
> ```
> 
> Kinda worked pretty good to increase coverage without causing side effects and added documentation as well 😅

Good idea! https://github.com/BSC-ES/autosubmit/issues/2585
